src/pages/Lending/Pool/components/Repay/RepayInput.tsx

```diff
@@ -332,6 +334,15 @@ export const RepayInput = ({
     setRepaymentAsset(collateralAsset)
   }, [collateralAsset, supportedRepaymentAssets, repaymentAsset, setRepaymentAsset])
 
+  // If the user disconnects the chain for the currently selected borrow asset, default to the collateral asset
+  useEffect(() => {
+    if (!collateralAsset || !repaymentAsset) return
+
+    if (!walletConnectedChainIds.includes(repaymentAsset.chainId)) {
+      setRepaymentAsset(collateralAsset)
+    }
+  }, [collateralAsset, repaymentAsset, setRepaymentAsset, walletConnectedChainIds])
+
```

Comment:
Yeah, agree that if the user intentionally disconnects both the repay/borrow account and the collateral account the user would have to select a different asset, but the aim here is to land on a sane default such that it's happy in the majority of cases, balancing effort with user benefit. If we think the balance of adding the business logic of multi-layered default for the case of users disconnecting multiple chains to avoid them clicking a different asset is worth it, happy to do so, but if not lets keep things simple.